enhancements/ingress/optional-ingress.md

```diff
@@ -0,0 +1,274 @@
+- Fully implement the ingress capability for the standalone OpenShift:
+  - Make the OpenShift installer tolerate the absence of the default ingress controller.
+  - Make the cluster authentication operator tolerate the absence of the default ingress controller.
+  - Make the cluster monitoring operator tolerate the absence of the default ingress controller.
+- Disable the Route API or try to tolerate the absence of the default ingress controller.
+- Implement the ingress operator disabling in the HyperShift's control plane operator.
+
+## Proposal
+
+### User Stories
+
+#### Disable the ingress on HyperShift
+
+The user of the HyperShift API wants to use an AWS ALB to route the traffic to its example application deployed on a hosted cluster.
+
+The user sets the capabilities in one of the HyperShift resources (**note**: the capability API is yet to be defined in HyperShift):
+```yaml
+apiVersion: hypershift.openshift.io/v1beta1
+kind: HostedCluster
+spec:
+  capabilities:
+    additionalDisabledCapabilities:
+    - ingress
+    baselineCapabilitySet: vCurrent
+...
+```
+
+The HyperShift's control plane operator skips the deployment of the ingress operator and creates a hosted cluster with the `ClusterVersion` as follows:
+```yaml
+apiVersion: config.openshift.io/v1
+kind: ClusterVersion
+metadata:
+  name: version
+spec:
+  capabilities:
+    additionalEnabledCapabilities:
+    - marketplace
+    - openshift-samples
+    - CSISnapshot
+    - Console
+    - Insights
+    - Storage
+    - baremetal
+    - marketplace
+    - openshift-samples
+    baselineCapabilitySet: None
+...
+```
+Note the missing `ingress` capability in the list of the additional enabled capabilities.
+
+To be able to route the traffic to its example application the user installs [AWS Load Balancer Operator](https://github.com/openshift/enhancements/blob/master/enhancements/ingress/aws-load-balancer-operator.md) and sets up the default `AWSLoadBalancerController`:
+```yaml
+apiVersion: networking.olm.openshift.io/v1
+kind: AWSLoadBalancerController
+metadata:
+  name: cluster
+spec:
+  credentials:
+    name: aws-sts-credentials
+```
+Then the user creates the `Ingress` resource for the default (`alb`) ingress class:
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: example
+  annotations:
+    alb.ingress.kubernetes.io/scheme: internet-facing
+spec:
+  ingressClass: alb
+  rules:
+  - host: example.com
+    http:
+        paths:
+        - path: /
+          backend:
+            service:
+              name: example
+              port:
+                number: 80
+```
+The aws load balancer controller will provision an ALB.
+
+Also, the user can install [External DNS Operator](https://github.com/openshift/enhancements/blob/master/enhancements/dns/externaldns-operator.md) to set up a convenient DNS name for its service (ingress support is coming soon).
+
+### Workflow Description
+TBD
```

Comment:
Some of what you have above in the user story section could fit here as the workflow, and then the user story could be just "as a $role, I want to disable the default ingress operator and use ALB instead".

Reply:
"as a $role ..." user stories and workflow description added.